### DISCUSSION TITLE
How can I search through a collection ?

### DISCUSSION THREAD

**Discussion by unripper** [appwrite/appwrite#3931] [Q&A] [answered]
I have an issue, I'm building a blogging app. The collection of `posts` looks like this

|  Attribute |      Type      |
|----------|:-------------:|
| title |  string |
| content |    string |

I need to search `fulltext` on this collection both title & content.

```typescript
appWriteSDK.database.listDocuments('posts', [
    Query.search('title', query),
    Query.search('content', query),
]);
```
How can I create index for this type of case ? 🤷‍♂️

**Comment by gepd** [comment, marked as answer]
Your index should looks like this:

![image](https://user-images.githubusercontent.com/7091609/166292631-34eb77b8-0c69-4131-aefa-37be7744f30c.png)

**Reply by alican92** [reply]
I'm having this issue. As I created the fulltext index with multiple attributes, it's not letting me search within the index or with the attribute name

**Comment by Haimantika** [comment]
Hi, @unripper is your issue solved? If yes, are we good to go ahead and close this?

**Comment by stnguyen90** [comment]
@unripper, if you need to search on *both* such that query can be in title *OR* content, I would recommend creating an additional attribute to store both title and content. Then, you would create a Fulltext index on that search attribute and query on that search attribute.

Is there anything else you need or can this be closed now?

**Reply by stnguyen90** [reply]
@unripper, would you please mark a response as an answer if it answers your question?

**Comment by xuala69** [comment]
I'm searching multiple(2 attributes) from Flutter which are indexed on the server. However, I am getting an error response as below 

```
AppwriteException: null, {"message":"Error: Error handler had an error","code":500,"file":"\/usr\/src\/code\/vendor\/utopia-php\/framework\/src\/App.php","line":610,"trace":[{"file":"\/usr\/src\/code\/vendor\/utopia-php\/framework\/src\/App.php","line":718,"function":"execute","class":"Utopia\\App","type":"->","args":[{},{}]},{"file":"\/usr\/src\/code\/app\/http.php","line":266,"function":"run","class":"Utopia\\App","type":"->","args":[{},{}]}],"version":"1.0.1"} (0)
```

**Reply by stnguyen90** [reply]
If you update to a newer version of Appwrite, the error should be clearer. The problem is most likely due to a missing index. You may need an index for each of the attributes as well.
